ezprefs.lua

```diff
@@ -0,0 +1,966 @@
+            cmd_counter = cmd_counter + 1
+
+            if reset and reset_pending then                             --put this in the first time only, or we'll end up
+                assign_cmd[prefname][cmd_counter] = "-reset"            --with nothing
+                cmd_counter = cmd_counter + 1
+                if not opts.silent then
+                    print("Clearing existing preferences.")
+                end
+                reset_pending = false
+            end
+
+            assign_cmd[prefname][cmd_counter] = "-unit"
+            cmd_counter = cmd_counter + 1
+            assign_cmd[prefname][cmd_counter] = tostring(unit)
+            cmd_counter = cmd_counter + 1
+
+            assign_cmd[prefname][cmd_counter] = "-" .. pref_types['output_args'][i]     --assign_cmd['material'][2-5,depending] = "-likematerial"
+            cmd_counter = cmd_counter + 1
+            assign_cmd[prefname][cmd_counter] = "["
+            cmd_counter = cmd_counter + 1
+            for _,v in ipairs(preferences[prefname]) do
+                table.insert(assign_cmd[prefname],cmd_counter,v)                        --adding the preferences
+                cmd_counter = cmd_counter + 1
+                prefcounter = prefcounter + 1
+            end
+            assign_cmd[prefname][cmd_counter] = "]"
+            cmd_counter = cmd_counter + 1
+
+            dfhack.run_script(table.unpack(assign_cmd[prefname]))
+        end
+    end
+    local unitthing = df.unit.find(unit)
+    local unitname = dfhack.df2console(dfhack.TranslateName(dfhack.units.getVisibleName(unitthing)))
+    if not opts.silent then
+        print("Added " .. prefcounter .. " preferences to " .. unitname .. ".")
+    end
+    return
+end
+-------------------------- DO_ACTIONS -----------------------------
+local function do_actions(job_list)
+    if opts.all then
+        --opts.silent = true
+        if opts.action == "jobs" then
+            for _,v in ipairs(opts.allunits) do
+                do_prefs(job_list,v,true,true)
+            end
+        elseif opts.action == "addjobs" then
+            for _,v in ipairs(opts.allunits) do
+                do_prefs(job_list,v,opts.reset,false)
+            end
+        else
+            print_yellow("You should not have gotten here.")
+        end
+    else
+        if opts.action == "jobs" then
+            do_prefs(job_list,opts.unit,true,true)
+        elseif opts.action == "addjobs" then
+            do_prefs(job_list,opts.unit,opts.reset,false)
+        elseif opts.action == "preview" then
+            do_preview(job_list)
+        else
+            print_yellow("You should not have gotten here.")
+        end
+    end
+end
+-------------------------- MAIN -------------------------------
+
+local function main(...)
+    local positionals = process_args({...})
+
+    if positionals ~= nil and #positionals > 0 then
+        print(shorthelp)
+        return
+    end
+
+    if opts.help then
+        print(help)
+        return

```

Comment:
once the help text is in the .rst file, you can print the help with `print(dfhack.script_help())`